src/rust/bitbox02-rust/src/hww/api/cardano/address.rs

```diff
@@ -43,9 +43,8 @@ pub const ADDRESS_HASH_SIZE: usize = 28;
 fn decode_shelley_payment_address(params: &params::Params, address: &str) -> Result<Vec<u8>, ()> {
     let result =
         bech32::primitives::decode::CheckedHrpstring::new::<bech32::Bech32>(address).or(Err(()))?;
-    // TODO: use `result.hrp().as_str()` once bech32 has a new release.
-    let hrp: String = result.hrp().char_iter().collect();
-    if hrp != params.bech32_hrp_payment {
+    let hrp = result.hrp();
+    if hrp.as_str() != params.bech32_hrp_payment {
```

Comment:
```suggestion
    if result.hrp().as_str() != params.bech32_hrp_payment {
```

Reply:
Can't do this as as_str() returns a reference to something inside the hrp instance, so the hrp() value needs to stay alive :(.

Reply:
I see :+1: thanks